go/pkg/client/cas.go

```diff
@@ -1054,6 +1055,10 @@ func (c *Client) downloadSingle(ctx context.Context, dg digest.Digest, reqs map[
 	}
 	r := rs[0]
 	rs = rs[1:]
+	if len(rs) == 0 {
+		// Download on same context as the issuing request, to support proper cancellation.
```

Comment:
Would it be possible to set the right context before calling downloadSingle? I think its more clear to know that the function uses the context you passed to it and not potentially another context from a side channel.

Reply:
Agreed, done.